### DISCUSSION TITLE
Treating str as not Iterable/Sequence[str]

### DISCUSSION THREAD

**Discussion by AlexeyDmitriev** [microsoft/pyright#8222] [Ideas]
Hi, I understand that in the language str can be iterated and return strings.

However, in my experience most of the time somebody writes

```python
def f(strings: Sequence[str]):
   ...
```

they expect `f("abc")` to be rejected.

Have you thought making this an error in pyright? (maybe under feature flag)
As far as I understand, another type checker (pytype) went this way and they were happy about the result

**Comment by erictraut** [comment]
Yes, this has been discussed before. I'd prefer to solve this problem in the Python typing spec and/or typeshed stubs rather than hard coding a non-standard exception. Doing so creates problems, especially when different type checkers deviate in their handling.

I don't think the typing community has reached a consensus on a solution. The [python typing forum](https://discuss.python.org/c/typing) is a good place to discuss this.

**Reply by AlexeyDmitriev** [reply]
Relevant thread I found in case somebody is also interested
https://discuss.python.org/t/an-interesting-pytype-experiment-and-a-possible-extension-to-strings/23749
https://github.com/python/typing/issues/256#issuecomment-902369240

It looks like there a lot of people who find this a problem, but idea to fix typeshed find pushback from few people and not enough enthusiasm to move it.
Maybe I'll find power to start yet another discussion
